# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/src/FilterSupport.h
# Commit: 6ea4f3bba40f
# Full Hash: 6ea4f3bba40f2fdf5c4893e0eb76fee7d4e05094
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2025-07-05 09:22:06
# Regressor Bug: 1912897
# File Overlap Count: 1
# Description:
#   Bug 1912897 - Support blurring and shadows for Accelerated Canvas2D. r=aosmond
#   
#   This adds support via FilterNodeWebgl for accelerating specific filters in AC2D.
#   In particular, it adds support for accelerated blurring using a separable blur
#   filter. This is then used to implement the Gaussian Blur filter as well as shadows.
# ==============================================================================

diff -r c6d00b59234d -r 6ea4f3bba40f gfx/src/FilterSupport.h
--- a/gfx/src/FilterSupport.h	Sat Jul 05 00:10:54 2025 +0000
+++ b/gfx/src/FilterSupport.h	Sat Jul 05 04:41:59 2025 +0000
@@ -400,15 +400,15 @@
    * space coordinates. aRenderRect specifies the part of the filter output
    * that will be drawn at (0, 0) into the draw target aDT, subject to the
    * current transform on aDT but with no additional scaling.
-   * The source surfaces must match their corresponding rect in size.
+   * The source filter nodes must match their corresponding rect in size.
    * aAdditionalImages carries the images that are referenced by the
    * eImageInputIndex attribute on any image primitives in the filter.
    */
   static void RenderFilterDescription(
       DrawTarget* aDT, const FilterDescription& aFilter,
-      const Rect& aRenderRect, SourceSurface* aSourceGraphic,
-      const IntRect& aSourceGraphicRect, SourceSurface* aFillPaint,
-      const IntRect& aFillPaintRect, SourceSurface* aStrokePaint,
+      const Rect& aRenderRect, RefPtr<FilterNode> aSourceGraphic,
+      const IntRect& aSourceGraphicRect, RefPtr<FilterNode> aFillPaint,
+      const IntRect& aFillPaintRect, RefPtr<FilterNode> aStrokePaint,
       const IntRect& aStrokePaintRect,
       nsTArray<RefPtr<SourceSurface>>& aAdditionalImages,
       const Point& aDestPoint, const DrawOptions& aOptions = DrawOptions());
